### PR TITLE
Add node definition for Boiler that inherits from DI

### PR DESCRIPTION
**Pull request by dacolgit**
## Purpose
add first revision of Boiler2 nodeset2
add DI nodeset2
add simple plugin for boiler2 and DI. Nodes are loaded  by plcNodeManager and accessed by an Opc client
* ...

## Does this introduce a breaking change?

```
[ ] Yes
[x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: src/ModelCompiler.cmd [hunk 1]
```diff
@@ -1,10 +1,17 @@
 @echo off
```
Comment (luiscantero) [inline src/ModelCompiler.cmd:1]: I think this file became too complex. As it was, it was good for compilin Boiler # 1 and the simple events. I think we should revert and create another file called e.g. ModelCompilerNodeSet2.cmd and put that logic there.

I'm open for discussions.

**Comment by luiscantero** [comment]
>  * Copyright (c) 2005-2021 The OPC Foundation, Inc. All rights reserved.

Pls move to folder CompanionSpecs\DI

---
Refers to: src/DI/Opc.Ua.DI.Classes.cs:2 in ffaf1d7. [](commit_id = ffaf1d7dd8f4cd0ee04c0438c135f242a0d34637, deletion_comment = False)

**Comment by luiscantero** [comment]
> <?xml version="1.0" encoding="utf-8" ?>

Move to folder CompanionSpecs

---
Refers to: src/Opc.Ua.Di.NodeSet2.xml:1 in ffaf1d7. [](commit_id = ffaf1d7dd8f4cd0ee04c0438c135f242a0d34637, deletion_comment = False)

**Comment by luiscantero** [comment]
> <UANodeSet LastModified="2023-05-16T09:46:08.767Z" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://opcfoundation.org/UA/2011/03/UANodeSet.xsd" xmlns:uax="http://opcfoundation.org/UA/2008/02/Types.xsd" xmlns:si="http://www.siemens.com/OPCUA/2017/SimaticNodeSetExtensions" xmlns:xsd="http://www.w3.org/2001/XMLSchema">

Move to folder Boilers. Rename to BoilerModel2.NodeSet2.xml

---
Refers to: src/Boiler.NodeSet2.xml:2 in ffaf1d7. [](commit_id = ffaf1d7dd8f4cd0ee04c0438c135f242a0d34637, deletion_comment = False)

File: src/Boilers/Boiler2/Build_ModelDesign.bat [hunk 2]
```diff
@@ -0,0 +1,4 @@
+@echo off
+
+REM Pass the model file name without .xml extension
+..\..\ModelCompiler.cmd ..\..\Boiler.NodeSet2 OpcPlc.Boiler2 Boiler2 ..\..\Opc.Ua.Di.NodeSet2 Opc.Ua.DI OpcUaDI
```
Comment (luiscantero) [inline src/Boilers/Boiler2/Build_ModelDesign.bat:4]: >OpcPlc.Boiler2 [](http://example.com/codeflow?start=46&length=14)

BoilerModel2

Comment (luiscantero) [inline src/Boilers/Boiler2/Build_ModelDesign.bat:4]: >ModelCompiler [](http://example.com/codeflow?start=6&length=13)

Call ModelCompilerNodeSet2.cmd

File: src/DI/Build_ModelDesign.bat [hunk 3]
```diff
@@ -0,0 +1,4 @@
+@echo off
+
+REM Pass the model file name without .xml extension
+..\ModelCompiler.cmd ..\Opc.Ua.Di.NodeSet2 Opc.Ua.DI OpcUaDI
```
Comment (luiscantero) [inline src/DI/Build_ModelDesign.bat:4]: >ModelCompiler [](http://example.com/codeflow?start=3&length=13)

Call ModelCompilerNodeSet2.cmd

File: src/PluginNodes/DIPluginNodes.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        {
+        }
+
+        private void AddNodes(FolderState methodsFolder)
+        {
+            // Load complex types from binary uanodes file.
+            _plcNodeManager.LoadPredefinedNodes(LoadPredefinedNodes);
+
+            // Find the Boiler1 node that was created when the model was loaded.
+            //var passiveNode = (BaseObjectState)_plcNodeManager.FindPredefinedNode(new NodeId(BoilerModel1.Objects.Boiler1, _plcNodeManager.NamespaceIndexes[(int)NamespaceType.Boiler]), typeof(BaseObjectState));
```
Comment (luiscantero) [inline src/PluginNodes/DIPluginNodes.cs:50]: >passiveNode [](http://example.com/codeflow?start=18&length=11)

Remove commented code

File: src/PluginNodes/DIPluginNodes.cs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            var passiveNode = (BaseObjectState)_plcNodeManager.FindPredefinedNode(new NodeId(BoilerModel1.Objects.Boiler1, _plcNodeManager.NamespaceIndexes[(int)NamespaceType.Boiler]), typeof(BaseObjectState));
```
Comment (luiscantero) [inline src/PluginNodes/DIPluginNodes.cs:51]: >BoilerModel1 [](http://example.com/codeflow?start=93&length=12)

Remove unused code

File: src/PluginNodes/DIPluginNodes.cs [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+        /// <summary>
+        /// Loads a node set from a file or resource and adds them to the set of predefined nodes.
+        /// </summary>
+        private NodeStateCollection LoadPredefinedNodes(ISystemContext context)
+        {
+            var predefinedNodes = new NodeStateCollection();
+
+            predefinedNodes.LoadFromBinaryResource(context,
+                "DI/Opc.Ua.DI.PredefinedNodes.uanodes", // CopyToOutputDirectory -> PreserveNewest.
```
Comment (luiscantero) [inline src/PluginNodes/DIPluginNodes.cs:62]: >DI [](http://example.com/codeflow?start=17&length=2)

CompanionSpecs/DI/...

File: src/PluginNodes/ZBoiler2PluginNodes.cs [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        {
+        }
+
+        private void AddNodes(FolderState methodsFolder)
+        {
+            // Load complex types from binary uanodes file.
+            _plcNodeManager.LoadPredefinedNodes(LoadPredefinedNodes);
+
+            // Find the Boiler1 node that was created when the model was loaded.
+            //var passiveNode = (BaseObjectState)_plcNodeManager.FindPredefinedNode(new NodeId(BoilerModel1.Objects.Boiler1, _plcNodeManager.NamespaceIndexes[(int)NamespaceType.Boiler]), typeof(BaseObjectState));
```
Comment (luiscantero) [inline src/PluginNodes/ZBoiler2PluginNodes.cs:51]: >//var passiveNode = (BaseObjectState)_plcNodeManager.FindPredefinedNode [](http://example.com/codeflow?start=12&length=71)

Remove commented line etc.

File: src/PluginNodes/ZBoiler2PluginNodes.cs [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpcPlc.PluginNodes
+{
+    public class ZBoiler2PluginNodes : IPluginNodes
```
Comment (luiscantero) [inline src/PluginNodes/ZBoiler2PluginNodes.cs:13]: >public [](http://example.com/codeflow?start=4&length=6)

Add summary

Comment (luiscantero) [inline src/PluginNodes/ZBoiler2PluginNodes.cs:13]: >ZBoiler2PluginNodes [](http://example.com/codeflow?start=17&length=19)

Remove Z

File: src/Boilers/Boiler2/Build_ModelDesign.bat (hunk 2 above)
Comment (dacolgit) [reply src/Boilers/Boiler2/Build_ModelDesign.bat:4]: changed

File: src/DI/Build_ModelDesign.bat (hunk 3 above)
Comment (dacolgit) [reply src/DI/Build_ModelDesign.bat:4]: changed

File: src/ModelCompilerNodeSet2.cmd [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+rem Display the results
+echo Model1 Filename : %filename1%
+echo Model2 Filename : %filename2%
+mkdir temp
+
+IF "%2" == "" (
+    echo Building %modelName1%.xml ...
+    %MODELCOMPILER% compile -version v104 -d2 "%MODELROOT%/%modelName1%.xml" -cg "%MODELROOT%/%modelName1%.csv" -o2 "%MODELROOT%/"
+) ELSE IF "%4" == "" (
```
Comment (luiscantero) [inline src/ModelCompilerNodeSet2.cmd:45]: I think we can remove this if contained in the original ModelCompiler.cmd

File: src/PluginNodes/Boiler2PluginNodes.cs [hunk 10]
```diff
@@ -0,0 +1,73 @@
+﻿namespace OpcPlc.PluginNodes;
+
+using BoilerModel2;
+using Opc.Ua;
+using OpcPlc.PluginNodes.Models;
+using System.Collections.Generic;
+using System.Reflection;
+
+/// <summary>
+/// Boiler node that inherit from DI.
```
Comment (luiscantero) [inline src/PluginNodes/Boiler2PluginNodes.cs:10]: >inherit [](http://example.com/codeflow?start=21&length=7)

inherits

File: src/ModelCompilerNodeSet2.cmd (hunk 9 above)
Comment (dacolgit) [reply src/ModelCompilerNodeSet2.cmd:45]: yes removed

File: src/PluginNodes/Boiler2PluginNodes.cs (hunk 10 above)
Comment (dacolgit) [reply src/PluginNodes/Boiler2PluginNodes.cs:10]: thanks fixed

**Pull request merged**